oshi-core/src/main/java/oshi/hardware/HardwareAbstractionLayer.java

```diff
@@ -77,13 +77,22 @@
     List<HWDiskStore> getDiskStores();
 
     /**
-     * Gets a list of {@link NetworkIF} objects, representing a network interface
+     * Gets a list of inet {@link NetworkIF} objects, representing a network interface
+     * The list excludes local interfaces.
      *
      * @return An {@code UnmodifiableList} of {@link NetworkIF} objects representing
      *         the interfaces
      */
     List<NetworkIF> getNetworkIFs();
 
+    /**
+     * Gets a list of all {@link NetworkIF} objects, representing a network interface
+     *
+     * @return An {@code UnmodifiableList} of {@link NetworkIF} objects representing
+     *         the interfaces
+     */
+    List<NetworkIF> getAllNetworkIFs();
```

Comment:
Instead of a new name, let's just overload the original one.  Make this second method `getNetworkIFs(boolean includeLocalAndVirtual);`.   Calling the first method will simply redirect to `getNetworkIFs(false)` and then you only have to really write the implementations for one method.

Reply:
I did not overload the method on purpose, following "Clean Code"'s statement about boolean parameters: https://medium.com/@amlcurran/clean-code-the-curse-of-a-boolean-parameter-c237a830b7a3

If you want, I can change it, though.

Reply:
Thanks for that article link.  I did spend some time reading it and considering this issue, and present the following counter for discussion.

While the article makes great points, it's dealing primarily with mutually exclusive control flows.  That, however, is not the case here. We are using overlapping/nearly-identical flows (iterating interfaces), with the only difference being a filtering parameter.  We are not making the code any more complex than it already is; the existing code includes this conditional:
```
if (!netint.isLoopback() && netint.getHardwareAddress() != null) { .... }
```
All you'd do is add the boolean parameter (or its opposite) to that conditional.

Another consideration is API consistency.  We use exactly the pattern I'm proposing in iterating Filesystems (see `FileSystem.getFileStores(boolean localOnly)`).   Of course, that's now complicated by the fact that there's a request to not filter the overlay filesystem types ( #1339 ), so perhaps a boolean is a poor choice there, or we may need a second parameter (probably a collection of strings).   Also, the `OperatingSystem` class has a default `getProcesses()` which gets all processes, delegating to `getProcesses(int limit, ProcessSort sort)` for the filtering/sorting/size limiting output.  I see this as a very similar type of application.

More importantly, the API is designed to transparently work with the Jackson project's `ObjectMapper` (see the `Json` class in the `oshi-demo` project).  By default, this class automatically, via reflection, assumes a `get*()` method maps directly to a field.  Perhaps using `get*()` is a bad choice to begin with, but at this point we're stuck with the API. Having two no-arg `get*()` methods will duplicate the JSON output.  The mapper does ignore the "getters" with arguments, and would ignore a method name that doesn't look like a getter (which is the reason `queryNetworkInterfaces()` is not `getNetworkInterfaces()`.)  

So I'm not overly opposed to you using two separate methods if you make the "full method" a `query*()` one, and then filter from those results in the `get*()` method.  That'd be an implementation detail.   However, given the similarity to the existing API implementations, I'd prefer to just have a boolean filter parameter on whether to include the loopback interfaces.

Reply:
Also, complicating things, in the existing method there are two filters, the loopback and the virtual (whether a hardware address exists).  So perhaps boolean isn't the best filtering, but open to ideas on how to make this both flexible and not overly complex (I don't want two booleans).  I do think an "unfiltered" output vs. a "most common filters" with a boolean switch is a good compromise.